--- BackupPC-3.3.1/lib/BackupPC/CGI/Browse.pm.orig	2016-07-15 10:32:24.007070271 +0200
+++ BackupPC-3.3.1/lib/BackupPC/CGI/Browse.pm	2016-07-15 10:33:00.920213213 +0200
@@ -65,7 +65,7 @@
     #
     # default to the newest backup
     #
-    if ( !defined($In{num}) && defined(@Backups) && @Backups > 0 ) {
+    if ( !defined($In{num}) && @Backups > 0 ) {
         $i = @Backups - 1;
         $num = $Backups[$i]{num};
     }
